### PR TITLE
NetSim: Animate packet sending

### PR DESCRIPTION
**Pull request by islemaster**
When you hit "Send" in packet-granularity mode, you can watch the binary drain out of the packet editor at your current bit rate!

out

Just to be clear, the "sending one bit at a time" is an illusion at this point.  We animate sending the packet, and when the animation is done we _actually_ send the packet.  I might need to reverse this eventually (so that the recipient can start animating receipt when the sender starts animating send) but that's easy-ish to change.  We do send each packet as it finishes animating, so packets are sent further apart at lower bitrates.

Adds a bitrate slider which is tied to the same property as the pulse rate slider under the hood, but they operate on dramatically different scales.  A real level should probably not display both of them, but things don't actually break if you do.

The send panel all disables while you're sending the packets, but you can change the bitrate while sending which will speed up/slow down the send animation.

Adds level config options for showing/setting/locking the bitrate slider and also retroactively for the chunk size slider.

### CODE REVIEW

**Comment by islemaster** [comment]
No ready to pull - I realized I missed a serialization step for a couple of the new level config options.

**Comment by islemaster** [comment]
Here's another sample screencast.  This one shows that you can crank the bitrate up to "unlimited" to instantly complete sending the current packet.
out

File: apps/src/netsim/NetSimMetronome.js [hunk 1]
```diff
@@ -72,6 +72,13 @@ NetSimMetronome.prototype.tick = function (clock) {
     this.lastPulseTime_ = clock.time;
   }
 
+  // An infinite interval means we're effectively paused, so snap to "empty"
```
Comment (davidsbailey) [inline apps/src/netsim/NetSimMetronome.js]: what does 'snap to "empty" ' mean?

Comment (islemaster) [inline apps/src/netsim/NetSimMetronome.js]: Good catch, that's not very descriptive.  Maybe it should say "snap to zero progress."  It looks like this:
empty_metronome

File: apps/src/netsim/NetSimPulseRateControl.js [hunk 2]
```diff
@@ -45,10 +45,11 @@ NetSimPulseRateControl.inherits(NetSimSlider.DecimalPrecisionSlider);
  * @override
  */
 NetSimPulseRateControl.prototype.valueToLabel = function (val) {
-  if (val === 1) {
-    return i18n.xSecondPerPulse({ x: val });
+  var rounded = Math.floor(val * 100) / 100;
+  if (rounded === 1) {
+    return i18n.xSecondPerPulse({ x: rounded });
```
Comment (davidsbailey) [inline apps/src/netsim/NetSimPulseRateControl.js]: Just FYI: some languages actually have more cardinalities than "one" and "many" (e.g. Polish, see http://www.unicode.org/cldr/charts/25/supplemental/language_plural_rules.html), however supporting this is not in scope for any of our products according to Tanya.

**Comment by davidsbailey** [comment]
LGTM

**Pull request merged**